cmd/peer-rest-server.go

```diff
@@ -1384,6 +1385,82 @@ func (s *peerRESTServer) DriveSpeedTestHandler(w http.ResponseWriter, r *http.Re
 	logger.LogIf(r.Context(), gob.NewEncoder(w).Encode(result))
 }
 
+// DevNull - everything goes to io.Discard
+func (s *peerRESTServer) DevNull(w http.ResponseWriter, r *http.Request) {
+	if !s.IsValid(w, r) {
+		s.writeErrorResponse(w, errors.New("invalid request"))
+		return
+	}
+	for {
+		n, err := io.CopyN(io.Discard, r.Body, 128*1024)
+		atomic.AddUint64(&globalNetperfRX, uint64(n))
+		if err != nil {
+			break
+		}
+	}
+}
+
+// Reader to read random data.
+type netperfReader struct {
+	eof bool
+	n   uint64
+	buf []byte
+}
+
+func (m *netperfReader) Read(b []byte) (int, error) {
+	if m.eof {
+		return 0, io.EOF
+	}
+	n := copy(b, m.buf)
+	atomic.AddUint64(&m.n, uint64(n))
+	return n, nil
+}
+
+func netperf(ctx context.Context, duration time.Duration) madmin.NetperfNodeResult {
+	globalNetperfRX = 0
+
+	r := &netperfReader{}
+	r.buf = make([]byte, 128*1024)
+	rand.Read(r.buf)
+
+	var wg sync.WaitGroup
+	for index := range globalNotificationSys.peerClients {
+		if globalNotificationSys.peerClients[index] == nil {
```

Comment:
If the peer is offline we should return an error for that

Reply:
when each server does netperf with peers, it does not maintain throughput numbers with each of the peers, instead it maintains aggregate throughput. So if one peer is down, we still test with other servers and report aggregate throughput. If in future we decide to show per peer throughput as well then we can report this error. Currently implementation and data structures are quite simple.